src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -301,12 +380,77 @@ public void setBaitSetName(final String name) {
             this.metrics.PROBE_SET = name;
         }
 
+        private int getBasesAtMinimumQualityInBlock(final SAMRecord record, final AlignmentBlock block) {
+            int basesInBLockAtMinimumQuality = 0;
+            final byte[] baseQualities = record.getBaseQualities();
+            for (int idx = block.getReadStart(); idx <= CoordMath.getEnd(block.getLength(), block.getReadStart()); idx++) { // idx is one-based
+                if (minimumBaseQuality <= baseQualities[idx-1]) basesInBLockAtMinimumQuality++;
+            }
+            return basesInBLockAtMinimumQuality;
+        }
+
+        private SAMRecord clipOverlappingBases(final SAMRecord rec) {
+            // NB: ignores how to handle supplemental records when present for both ends by just using the mate information in the record.
+
+            if (!rec.getReadPairedFlag()) return rec;
+
+            // Only clip records that are left-most in genomic order and overlapping and are the first end
```

Comment:
I think this is wrong.  If you have an FR read pair where the R2 is earlier on the genome on the F strand and R1 is later on the genome on the R strand, and they overlap, won't this erroneously skip clipping?  I think the clipping logic shouldn't care if it's R1 or R2.